### DISCUSSION TITLE
Cant understand example!?

### DISCUSSION THREAD

**Discussion by zhuravkov** [SeaQL/sea-query#711] [Q&A]
Hi, in this exaple https://www.sea-ql.org/SeaORM/docs/migration/writing-migration/#schema-creation-methods 
 I can't understand how the code works, what we try to do

```
#[derive(Iden, EnumIter)]
pub enum Category {
    Table,
    #[iden = "Feed"]
    Feed,
    #[iden = "Story"]
    Story,
}
```
on new DB POSTGRES i get error<< Execution Error: error returned from database: type "category" does not exist>>

**Comment by tyt2y3** [comment]
If you are using Postgres you need to create the type in the database first (which is in the same section).

Like:

```rust
Type::create().as_enum(Category::Table)...
```

**Reply by zhuravkov** [reply]
Thanks, I think I understand, first you need to perform the tax 
```
CREATE TYPE category AS ENUM ('Feed, 'Story')
```

like this:
```
pub struct Migration;

#[async_trait::async_trait]
impl MigrationTrait for Migration {
    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
        manager
            .create_type(
                Type::create()
                    .as_enum(Category::Table)
                    .values([Category::Feed, Category::Story])
                    .to_owned(),
            )
            .await
     }

}

#[derive(Iden, EnumIter)]
pub enum Category {
    Table,
    #[iden = "Feed"]
    Feed,
    #[iden = "Story"]
    Story,
}
```
and then https://www.sea-ql.org/SeaORM/docs/migration/writing-migration/#schema-creation-methods do something like this:
```
 CREATE TABLE Post (
   .....
    category category
);
```
Then sea-orm-cli generate entity  -o entity/src   create: 
```
pub enum Category {
    #[sea_orm(string_value = "Feed")]
    Feed,
    #[sea_orm(string_value = "Story")]
    Story,
}
``` 
it's greate!!
But the last thing I can’t understand is how this enumeration affects the code:
```
                .col(
                    ColumnDef::new(Post::Category)
                        .enumeration(Category::Table, [Category::Feed])
```
 if I don't pass all the values ​​into the array, how will this affect the table?
